static/elements/chromedash-feature.html

```diff
@@ -1,7 +1,6 @@
 <link rel="import" href="../bower_components/polymer/polymer.html">
 <link rel="import" href="../bower_components/iron-icon/iron-icon.html">
 <link rel="import" href="../bower_components/paper-styles/color.html">
-<link rel="import" href="chromedash-color-status.html">
```

Comment:
You're removing the HTML import, but not adding a JS `import` or `<script type="module">`. How are you getting `chromedash-color-status` into here?

Reply:
All lit-element components (currently only chromedash-color-status) are bundled by rollup into one .js file, and added to the page base template (base.html and blank.html).

From this and the script comment, I guess your expectation is: modules are not bundled together, and loaded in each template they are being used. Is it?

Reply:
Every file that depends on another should directly import it. If you don't model the dependency graph accurately, code-splitting bundlers will not be able to work correctly.

I would actually recommend that you add Rollup last in this process. You should be able to get very far without bundling anything, since all the current browsers support modules.

Reply:
Sounds good, thanks!

Can you give me some advice on the process - "add Rollup last in this process"?
My understanding are:
- The usage of rollup in our current setup are: 1) handle import lit-element. 2) combine all components into one .js file. Now it seems 2) is not a good practice.
- Since the components are lit-element, they need to be bundled with lit-element.

So what I should do now are:
MyComponent (using lit-element) => bundle with lit-element, export MyComponentBundled. Same for all components. But how should I do this bundling? (If I use rollup/webpack, they always do the (2) above, right?)

Reply:
I'm coming from the perspective that a bundler should only take a working project as input and output an optimized working projects. LitElement doesn't need to be bundled with components that use it - LitElement will load when the component imports it. The only thing we need to do right now is translate bare import specifiers to URLs. Rollup does this with the resolve plugin, but there are dev servers like Polymer CLI's server or [open-wc.org's dev server](https://open-wc.org/developing/owc-dev-server.html) that do this on the fly.

Bundling into a single file is perfectly fine in production, but it shouldn't be required for development, and one reason to structure your imports so that every file imports only the direct dependencies they use is so that the project will load without bundling. (another is so IDEs and other tools can understand a module better)

> MyComponent (using lit-element) => bundle with lit-element, export MyComponentBundled. Same for all components.

I definitely would not bundle LitElement into every component. That will severely bloat the bundles.

What I would do is just get the project working without any bundling. Then analyze whan bundles you need based on the entrypoints. A good bundler will be able to handle multiple entry points and build multiple bundles, with shared bundles for the shared libraries.

Reply:
Ah, Okay. I'll try this method. Thanks!